src/s_y.rs

```diff
@@ -214,11 +214,36 @@ pub fn calculate(input: &Vec<Token>) -> Result<f32, Vec<String>> {
             Token::WholeNumber(n) => stack.push(Token::DecimalNumber(n as f32)),
             Token::FunctionCall(function_name) => {
                 match &function_name as &str {
-                    "cos" => {
+                    "log" => {
                         let arg = stack.pop();
-
-                        if let Some(Token::DecimalNumber(n1)) = arg {
-                            stack.push(Token::DecimalNumber(n1.cos()));
+                        match arg {
+                            Some(Token::DecimalNumber(n)) => stack.push(Token::DecimalNumber((n as f32).log10())),
+                            Some(Token::WholeNumber(n)) => stack.push(Token::DecimalNumber((n as f32).log10())),
```

Comment:
do we want to make a second log function for things other than base 10?/

Reply:
Sure, there's a function for it, I just wasn't sure if a TI could do that. Would this function then be called log10?

Reply:
yeah, the nomenclature on a ti is log(n, base) we should probably stick to log 10 for now as it requires a function of multiple arguments which the shunting yard doesnt have yet, add it as a //TODO: